credoai/reporting/dataset_base.py

```diff
@@ -0,0 +1,185 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+from credoai.reporting.credo_report import CredoReport
+from credoai.reporting import plot_utils
+
+
+class DatasetModuleReport(CredoReport):
+    def __init__(self, module):
+        super().__init__()
+        self.module = module
+
+    def create_report(self, filename=None):
+        """Creates a fairness report for dataset assessment module
+
+        Parameters
+        ----------
+        filename : string, optional
+            If given, the location where the generated pdf report will be saved, by default Non
+
+        Returns
+        -------
+        array of figures
+        """
+        results_all = self.module.get_results()
+
+        # Generate sample balance barplots
+        fig, axs = plt.subplots(nrows=3, figsize=(8, 8), dpi=150)
+        results = results_all["balance_metrics"]["sample_balance"]
+        df = pd.DataFrame(results)
+        sensitive_feature_name = list(df.drop(["count", "percentage"], axis=1).columns)[
+            0
+        ]
+
+        ax = sns.barplot(
+            x="count",
+            y=sensitive_feature_name,
+            data=df,
+            palette=plot_utils.credo_diverging_palette(1),
+            alpha=1,
+            ax=axs[0],
+        )
+        fig.patch.set_facecolor("white")
+        ax.set_frame_on(False)
+        ax.set_title("Data balance across " + sensitive_feature_name + " subgroups")
+        ax.set_xlabel("Number of data samples")
+        ax.set_ylabel("")
+
+        # Generate label balance barplots
+        results = results_all["balance_metrics"]["label_balance"]
+        df = pd.DataFrame(results)
+        label_name = list(df.drop([sensitive_feature_name, "count"], axis=1).columns)[0]
+
+        num_classes = df[label_name].nunique()
+        ax = sns.barplot(
+            x="count",
+            y=sensitive_feature_name,
+            hue=label_name,
+            data=df,
+            palette=plot_utils.credo_diverging_palette(num_classes),
+            alpha=1,
+            ax=axs[1],
+        )
+        fig.patch.set_facecolor("white")
+        ax.set_frame_on(False)
```

Comment:
I've changed these to `sns.despine`. In general I think it's better (not always) to have the axis lines on the left and bottom